### PR TITLE
Update TeamsInfo to support CloudAdapter

### LINKED ISSUE #5685: TeamsInfo should work with CloudAdapter
## Version
main branch

## Describe the bug
SendMessageToTeamsChannelAsync is currently casting the adapter to **BotFrameworkAdapter** `((BotFrameworkAdapter)turnContext.Adapter).CreateConversationAsync`

https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder/Teams/TeamsInfo.cs#L233

This should be updated to work with CloudAdapter since it now also supports CreateConversation.

customer issue reference:
https://github.com/microsoft/botframework-components/issues/1150

### PR DESCRIPTION
**Pull request by johnataylor**
fixes #5685

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder/Teams/TeamsInfo.cs [hunk 1]
```diff
@@ -246,6 +246,59 @@ public static Task<TeamsChannelAccount> GetMemberAsync(ITurnContext turnContext,
             return new Tuple<ConversationReference, string>(conversationReference, newActivityId);
         }
 
+        /// <summary>
+        /// Creates a new thread in a team chat and sends an activity to that new thread.
+        /// </summary>
+        /// <param name="turnContext"> Turn context. </param>
```
Comment (EricDahlvang) [inline libraries/Microsoft.Bot.Builder/Teams/TeamsInfo.cs:252]: nit: spacing looks odd before/after the param comments here

**Review by EricDahlvang — APPROVED** [review]
We might need to revisit this when we add support for create conversation from a skill. This will probably require audience.

**Review by clearab — CHANGES_REQUESTED** [review]
I think the descriptions got mixed up - the description for activity is the channel ID, and channel ID has a fragment of "cancellation token." (in the code comments)

**Comment by johnataylor** [comment]
Thanks @clearab nice catch. That must have been there for ages - I updated the old functions description too.

**Review by mrivera-ms — CHANGES_REQUESTED** [review]
:clock1:

**Review by mrivera-ms — APPROVED** [review]
:shipit:

**Review by clearab — APPROVED** [review]

**Pull request merged**